% -*-latex-*-
%\documentclass[pdftex,handout]{beamer}
\documentclass[pdftex,12pt]{beamer}
\usepackage{etex}
\let\latexput\put
\usepackage{pictex}
\let\pictexput\put
\let\put\latexput
\newcommand{\mutation}{$\cdot$}
\usepackage{ifthen}
\newcounter{showthis}
\setcounter{showthis}{1}
\newdimen\thusfar
\newdimen\plotht
\newdimen\plotwd
\newdimen\plotsp
\newdimen\xunit
\newdimen\yunit
\newdimen\offsety
\setbeamercovered{transparent}
\setbeamertemplate{footline}[frame number]
\author{Alan R. Rogers}
\date{\today}
\begin{document}

\title{Site Patterns and Population History: the Intuition that
  Underlies Legofit}
\frame{\titlepage}

\begin{frame}
  \frametitle{Notation for populations}
  \begin{columns}
    \column{0.3\textwidth}
    \input{fignotation}
    \column{0.7\textwidth}
    \raggedleft
    $X$, $Y$, and $N$ are populations: African, European, and Neanderthal.

    \bigskip
    
    \textit{XY}: population ancestral to $X$ and $Y$.

    \bigskip
    
    \textit{XYN}: ancestral to $X$, $Y$, and $N$.
    \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Nucleotide site patterns}
  \begin{columns}
    \column{0.3\textwidth}
    \input{figcongruent}
    \column{0.7\textwidth}
    \raggedleft
    Haploid sample: 1 nucleotide from each population.

    \bigskip
    
    Mutation on red segment would appear in samples from $X$, $Y$, not
    that from $N$.

    \bigskip
    Call this the \textit{xy} site pattern.
  \end{columns}
  \bigskip\raggedleft I will write $xy\succ yn$ to mean that $xy$ is more
  common than $yn$.
\end{frame}

\begin{frame}
  \frametitle{Why haploid samples?}
  No variation within populations $\Rightarrow$ results not affected
  by recent history of population size.

  \bigskip

  The haploid samples are hypothetical; our real samples are
  larger. We use all the genomes in the real data to calculate the
  probability of observing site pattern \emph{xy} in a hypothetical
  haploid sample:
  \[
  \Pr[xy] = p_X p_Y (1-p_N)
  \]
  where $p_i$ is the frequency of the derived allele in the sample
  from population $i$. These site pattern frequencies are our data.
\end{frame}  

\begin{frame}
  \frametitle{Calling ancestral and derived alleles}
{\centering\input{figancder}\\}
    \bigskip
    
    2 mutations needed if $C$ is ancestral.

    \bigskip

    Only 1 needed if $T$ is ancestral.

    \bigskip

    Prefer hypothesis requiring fewer mutations, because mutations
    are rare.
\end{frame}

\begin{frame}
  \frametitle{Observed site pattern frequencies}
  \begin{columns}
    \column{0.7\textwidth}
    \includegraphics[width=\linewidth]{xynd-frq.pdf}
    \column{0.3\textwidth}
    \raggedleft

    $X$, Africa; $Y$, Europe; $N$, Neanderthal; $D$, Denisovan.

    \bigskip

    Horizontal axis: rel.\ freq.\ of each site pattern

    \bigskip

    ``Dots'' w/i circles are 95\% confidence intervals.
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{The pattern in the data}
  \begin{columns}
    \column{0.7\textwidth}
    \includegraphics[width=\linewidth]{xynd-frq.pdf}
    \column{0.3\textwidth}
    \raggedleft

    Lots of singletons ($x$, $y$, $n$, and $d$)

    \bigskip

    Two doubletons (\textit{xy} and \textit{nd}) especially common

    \bigskip

    \textit{yn} more common than other rare doubletons.

    \bigskip

    \textit{xyn} more common than other tripletons.
  \end{columns}
  \bigskip\raggedleft
How can we understand this pattern?
\end{frame}

\begin{frame}
  \frametitle{1st pass: no frills}
  \begin{columns}
    \column{0.5\textwidth}
    \input{figEarlyND}
    \column{0.5\textwidth}
    \raggedleft
    No gene flow; gene genealogy matches population tree.

    \bigskip

    Many other genealogies are possible, but this one will be common.

    \bigskip

    Captures large-scale pattern; misses subtleties.
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Why are \textit{xy} and \textit{nd} so common?}
  \begin{columns}
    \column{0.5\textwidth}
    \input{figEarlyND}
    \column{0.5\textwidth}
    \raggedleft
    Mutation on blue $\rightarrow$ \textit{xy};
    mutation on red $\rightarrow$ \textit{nd}.

    \bigskip

    \textit{xy} and \textit{nd} are common because $X$ and $Y$ are
    closely related, as are $N$ and $D$.
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Why is $xy \succ nd$?}
  \begin{columns}
    \column{0.5\textwidth}
    \input{figEarlyND}
    \column{0.5\textwidth}
    \raggedleft
    Blue branch is longer than red, because $X$ and $Y$ separated more
    recently than $N$ and $D$.

    \bigskip

    Explains why $xy \succ nd$.
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Data again: \textit{xy} and \textit{nd} common, but
  $xy \succ nd$}
    {\centering\includegraphics[width=0.9\linewidth]{xynd-frq.pdf}\\}
\end{frame}

\begin{frame}
  \frametitle{An alternate hypothesis}
  \begin{columns}
    \column{0.5\textwidth}
    \input{figLargeND}
    \column{0.5\textwidth}
    \raggedleft
    Separation times are equal.

    \bigskip
    
    But $ND$ is large, so coalescence is slow, and red branch is short.

    \bigskip

    $xy \succ nd$ because $ND$ is larger than $XY$.

    \bigskip

    The two hypotheses are hard to tell apart.
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Counterintuitive site patterns}
  {\centering\input{figlinsort}\\}  
\end{frame}

\begin{frame}
  \frametitle{Incomplete lineage sorting}
  Suppose that, as we trace the ancestry of our sample backwards in
  time, the lineages from $X$ and $Y$ don't
  coalesce until we reach \textit{XYN}.

  \bigskip

  Then there are three lineages, $X$, $Y$, and $N$, in the same
  population.

  \bigskip

  They can coalesce in any order.

  \bigskip

  Site patterns \textit{xy}, \textit{xn}, and \textit{yn} are equally
  likely.

  \bigskip

  This process is called ``incomplete lineage sorting.''
\end{frame}  

\begin{frame}
  \frametitle{Pattern \textit{xy} can also arise another way}
  \begin{columns}
    \column{0.3\textwidth}
    \input{figcongruent}
    \column{0.7\textwidth}
    \raggedleft
    The lineages from $X$ and $Y$ may also coalesce w/i \textit{XY},
    generating site pattern \textit{xy}.

    \bigskip
    
    So $xy \succ xn, yn$.

    \bigskip

    \textit{xn} and \textit{yn} should be equally common.
  \end{columns}

  \bigskip\raggedleft
  
This is the pattern expected in the absence of gene flow.
\end{frame}

\begin{frame}
  \frametitle{Does incomplete lineage sorting (ILS) explain the data?}
  \begin{columns}
    \column{0.7\textwidth}
    \includegraphics[width=\linewidth]{xynd-frq.pdf}
    \column{0.3\textwidth}
    \raggedleft

    \textit{xd} and \textit{yd} have equal frequencies, as they should
    under ILS.

    \bigskip

    But $yn \succ xn$, and the difference $>$ the confidence intervals.

    \bigskip

    Why?

  \end{columns}
  \bigskip\raggedleft
\end{frame}

\begin{frame}
  \frametitle{The effect of gene flow}
  \begin{columns}
  \column{0.7\textwidth}
  \input{figmix}
  \column{0.3\textwidth}
  \raggedleft

  $N\rightarrow Y$ gene flow inflates the frequency of \textit{yn}.

  \bigskip

  Also inflates frequency of $x$.

  \bigskip

  Effects are small unless the rate of gene flow is high.
  \end{columns}
\end{frame}


\begin{frame}
  \frametitle{Data are consistent with $N \rightarrow Y$ gene flow}
  \begin{columns}
    \column{0.7\textwidth}
    \includegraphics[width=\linewidth]{xynd-frq.pdf}
    \column{0.3\textwidth}
    \raggedleft

    $yn \succ xn$, and $x \succ y$.

    \bigskip

    Signature of $N \rightarrow Y$ gene flow.
    
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Puzzling excess of $d$ site pattern}
  \begin{columns}
    \column{0.7\textwidth}
    \includegraphics[width=\linewidth]{xynd-frq.pdf}
    \column{0.3\textwidth}
      \raggedleft

      $d$ is most common singleton

      \bigskip

      Suggests Denisovan fossil is young and $N$-$D$ separation old.

      \bigskip

      But our 2017 analysis of this hypothesis led to absurd result:
      Denisovan fossil only 4000 y old.
  \end{columns}

  \bigskip\raggedleft
      
  Something was missing from our model
\end{frame}

\begin{frame}
  \frametitle{Admixture from superarchaics into Denisovans}
  \begin{columns}
    \column{0.7\textwidth}
    \input{fighyparch}
    \column{0.3\textwidth}
    \raggedleft
    $S$ is a ``superarchaic'' hominin, distantly related to all
    others.

    \bigskip

    $S \rightarrow D$ gene flow inflates frequency of $d$ and
    \textit{xyn}. 
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{Superarchaic gene flow into $D$}
  \begin{columns}
    \column{0.7\textwidth}
    \includegraphics[width=\linewidth]{xynd-frq.pdf}
    \column{0.3\textwidth}
      \raggedleft

      $d$ most common singleton

      \bigskip

      \textit{xyn} most common tripleton

      \bigskip

      Signature of $S \rightarrow D$ gene flow (Pr{\"u}fer et al 2014).
  \end{columns}
\end{frame}

\begin{frame}
  \frametitle{What we learned, just by staring at the data}
  \begin{enumerate}
    \item Europeans and Africans are close relatives.
    \item So were Neanderthals and Denisovans.
    \item European-African separation more recent than
      Neanderthal-Denisovan.
    \item Neanderthals contributed genes to Europeans
    \item Superarchaics contributed genes to Denisovans.
  \end{enumerate}

  \bigskip
  
This analysis has been exploratory. Legofit extends these ideas to
estimate parameters and test hypotheses about history.
\end{frame}  

\end{document}


\begin{frame}
\frametitle{}
\end{frame}
