# AGAR Workshop on Computational Genetics

Timothy Webster and Alan Rogers are hosting the AGAR Workshop on
Computational Genetics July 27-29, 2022. The workshop will focus on
methods for inferring population history and admixture. It will be
virtual, with lectures via Zoom and materials made available on
[github](https://github.com/alanrogers/agar22.git).

# How to apply

Applications are now closed. If you applied, you should have received
an invitation to join the agar22 Google Group, which we will be using
to communicate with participants.

# Schedule

We will meet each day from 9am-12pm and from 1-4pm, Mountain Daylight
Time (UTC-6).

<table>
<tr>
<th>Day</th>
<th>Time</th>
<th>Speaker</th>
<th>Topic</th>
</tr>

<tr>
<td>2022-7-27</td>
<td>AM</td>
<td>Ainash Childebayeva</td>
<td>Working with genomic data</td>
</tr>

<tr>
<td>2022-7-27</td>
<td>PM</td>
<td>Tim Webster</td>
<td>Reproducible pipelines; the command line</td>
</tr>

<tr>
<td>2022-7-28</td>
<td>AM</td>
<td>Colin Brand</td>
<td>Simulation with msprime</td>
</tr>

<tr>
<td>2022-7-28</td>
<td>PM</td>
<td>Shyamalika Gopalan</td>
<td>Introgression: background and theory</td>
</tr>

<tr>
<td>2022-7-29</td>
<td>AM</td>
<td>Robert Maier</td>
<td>Admixtools</td>
</tr>

<tr>
<td>2022-7-29</td>
<td>PM</td>
<td>Alan Rogers</td>
<td>Legofit</td>
</tr>
</table>

# Background reading

* [Just enough probability](http://content.csbs.utah.edu/~rogers/pubs/Rogers-JEP.pdf)

* Chs. 4-5 of [Lecture notes on gene genealogies](ggeneal.pdf)

* [Rogers 2022](https://arxiv.org/abs/2201.02668) Using genetic data
    to build intuition about population history.

* During the workshop, we will make use of the
[Bash shell](https://www.gnu.org/software/bash/manual), a command-line
interpreter that is widely used in computational genetics. It is
available on MacOS through the "Terminal" application and on Windows
through the
[Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install). Anyone
using Linux will be familiar with it already.  We won't try to teach
you Bash, but you should familiarize yourself with the basic
commands--"ls", "pwd", "cd", "rm", and "mkdir"--and also with bash
pipes, and the redirection of input and output. In addition to these
basic commands, please read up on
"[seq](https://linuxhandbook.com/seq-command/)" and
"[xargs](https://en.wikipedia.org/wiki/Xargs)", which are used
extensively in pipelines.

# Tutorials

* [Legofit](legofit/legotut.pdf)

* [Msprime](msprime/msptut.pdf)

# Faculty

* [Colin Brand](https://colinmbrand.weebly.com) is a postdoc in the
  [Capra Lab](https://http://www.capralab.org) at UCSF. His current
  research focuses on characterizing adaptation and introgression in non-human
  primates and predicting phenotypes from genomic data in archaic hominins.

* [Ainash Childebayeva](https://sph.umich.edu/stories/2020posts/ainash-childebayeva.html)
  is in the Dept of Archaeogenetics, in the
  [Max Planck Instute of Evolutionary Anthropology](https://pure.mpg.de/cone/persons/resource/persons242932).

* Shyamalika Gopalan is a postdoc in the
  [Goldberg lab](https://www.goldberglab.org/people) at Duke
  University. She has
  [extensive experience](https://scholar.google.com/citations?user=mZhqPRMAAAAJ&hl=en)
  using genetic data to infer population history.

* [Robert Maier](https://heb.fas.harvard.edu/people/robert-maier) is a
  Research Fellow at the Broad Institute, Harvard University. He is
  the lead developer of AdmixTools2, which he will teach during the
  workshop. 

* [Alan Rogers](https://anthro.utah.edu/profile.php?unid=u0028949) is
  in the Anthro Dept of the University of Utah. He has developed
  several methods for inferring population history from genetic
  data. He will cover one of those methods,
  [Legofit](https://alanrogers.github.io/legofit/html/index.html),
  during the workshop.

* [Timothy Webster](https://faculty.utah.edu/u6023206-TIM_WEBSTER/hm/index.hml)
  is in the Anthro Dept of the University of Utah. He uses genomic
  data and computational methods to understand primate evolution,
  ecology, and behavior. In particular, he explores (1)
  macroevolutionary processes, especially speciation, adaptation, and
  lineage-specific demographic history, (2) modern and historical
  aspects of behavioral ecology and social organization, and (3) sex
  chromosome evolution.

