# AGAR Workshop on Computational Genetics

Timothy Webster and Alan Rogers are hosting the AGAR Workshop on
Computational Genetics July 27-29, 2022. The workshop will focus on
methods for inferring population history and admixture. It will be
virtual, with lectures via Zoom and materials made available on
[github](https://github.com/alanrogers/agar22.git).

# How to apply

The workshop is free, but attendees must apply. Applications are due
*July 15, 2022*.

The workshop is open to anyone interested in using genetic data to
study population history. If space runs short, we will prioritize those
most likely to benefit from the workshop.

To apply, [please fill out this form](https://forms.gle/MoyVS7JsWc94SsSj9).

# Modules

1. Working with genomic data (types of genomic data, pros/cons, etc.).
Speaker to be announced.
<!--Invitations: Ainash Childebayeva.-->

2. Reproducible bioinformatic pipelines and working on the command
   line. [Tim Webster](https://www.websterlab.org/)

3. Simulation with msprime. Colin Brand.

4. Introgression: background and theory. Shyamalika Gopalan.

5. Admixtools. Robert Maier

6. Legofit. Alan Rogers

# Background reading

* [Just enough probability](http://content.csbs.utah.edu/~rogers/pubs/Rogers-JEP.pdf)

* Chs. 4-5 of [Lecture notes on gene genealogies](ggeneal.pdf)

* [Rogers 2022](https://arxiv.org/abs/2201.02668) Using genetic data
    to build intuition about population history.

* During the workshop, we will make use of the
[Bash shell](https://www.gnu.org/software/bash/manual), a command-line
interpreter that is widely used in computational genetics. It is
available on MacOS through the "Terminal" application and on Windows
through the
[Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install). Anyone
using Linux will be familiar with it already.  We won't try to teach
you Bash, but you should familiarize yourself with the basic
commands--"ls", "pwd", "cd", "rm", and "mkdir"--and also with bash
pipes, and the redirection of input and output.

# Tutorials

* [Legofit](legofit/legotut.pdf)

* [Msprime](msprime/msptut.pdf)

# Faculty

* Shyamalika Gopalan is a postdoc in the
  [Goldberg lab](https://www.goldberglab.org/people) at Duke
  University. She has
  [extensive experience](https://scholar.google.com/citations?user=mZhqPRMAAAAJ&hl=en)
  using genetic data to infer population history.

* [Robert Maier](https://heb.fas.harvard.edu/people/robert-maier) is a
  Research Fellow at the Broad Institute, Harvard University. He is
  the lead developer of AdmixTools2, which he will teach during the
  workshop. 

* [Alan Rogers](https://anthro.utah.edu/profile.php?unid=u0028949) is
  in the Anthro Dept of Anthropology of the University of Utah. He has
  developed several methods for inferring population history from
  genetic data. He will cover one of those methods,
  [Legofit](https://alanrogers.github.io/legofit/html/index.html),
  during the workshop.

* [Timothy Webster](https://faculty.utah.edu/u6023206-TIM_WEBSTER/hm/index.hml)
  is in the Anthro Dept of the University of Utah. He uses genomic
  data and computational methods to understand primate evolution,
  ecology, and behavior. In particular, he explores (1)
  macroevolutionary processes, especially speciation, adaptation, and
  lineage-specific demographic history, (2) modern and historical
  aspects of behavioral ecology and social organization, and (3) sex
  chromosome evolution.

